src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import './Dropdown.scss';
+import cn from 'classnames';
+import { debounce } from 'lodash';
+import { Person } from '../../types/Person';
+import { DropdownItem } from '../DropdownItem';
+
+type Props = {
+  people: Person[];
+  setSelectedPerson: (person: Person | null) => void;
+  delay: number;
+
+};
+
+export const Dropdown: React.FC<Props> = (
+  {
+    people,
+    setSelectedPerson,
+    delay,
+
+  },
+) => {
+  const [query, setQuery] = useState('');
+  const [debouncedQuery, setDebouncedQuery] = useState('');
+
+  const applyDebouncedQuery = useCallback(
+    debounce(setDebouncedQuery, delay),
+    [],
+  );
+
+  const visiblePeople: Person[] = useMemo(() => (
+    people.filter(
+      ({ name }) => name.toLowerCase().includes(debouncedQuery.toLowerCase()),
+    )
+  ), [debouncedQuery]);
+
+  const handleQueryChange = useCallback(
+    (newQuery: string) => {
+      setSelectedPerson(null);
+      setQuery(newQuery);
+      applyDebouncedQuery(newQuery.trim());
+    },
+    [applyDebouncedQuery],
+  );
+
+  const handlePersonClick = useCallback(
+    (person: Person) => {
+      setSelectedPerson(person);
+      setQuery(person.name);
+      setDebouncedQuery('');
+    },
+    [],
+  );
+
+  const handleResetQuery = useCallback(
+    () => {
+      setSelectedPerson(null);
+      setQuery('');
+      setDebouncedQuery('');
+    },
+    [],
+  );
```

Comment:
Why do you wrap each handler in useCallback? write why please

Reply:
To prevent extra rendering in smaller components in Dropdown if something will be changed inside Dropdown, but forgot to split it.